fvm/src/call_manager/default.rs

```diff
@@ -242,21 +247,83 @@ where
         self.nonce
     }
 
-    // Helper for creating actors. This really doesn't belong on this trait.
-
-    fn next_actor_idx(&mut self) -> u64 {
-        let ret = self.num_actors_created;
-        self.num_actors_created += 1;
-        ret
+    fn next_actor_address(&self) -> Address {
+        // Base the next address on the address specified as the message origin. This lets us use,
+        // e.g., an f2 address even if we can't look it up anywhere.
+        //
+        // Of course, if the user decides to send from an f0 address without waiting for finality,
+        // their "stable" address may not be as stable as they'd like. But that's their problem.
+        //
+        // In case you're wondering: but what if someone _else_ is relying on the stability of this
+        // address? They shouldn't be. The sender can always _replace_ a message with a new message,
+        // and completely change how f2 addresses are assigned. Only the message sender can rely on
+        // an f2 address (before finality).
+        let mut b = to_vec(&self.origin_address).expect("failed to serialize address");
+        b.extend_from_slice(&self.nonce.to_be_bytes());
+        b.extend_from_slice(&self.num_actors_created.to_be_bytes());
+        Address::new_actor(&b)
     }
 
-    fn invocation_count(&self) -> u64 {
-        self.invocation_count
+    fn create_actor(
```

Comment:
I moved this here because:

1. It doesn't actually care about the kernel.
2. We'll eventually need to change this to do an internal "send" to actually construct the actor (which is in the domain of the CallManager anyways).

Reply:
Note: this hasn't changed, just moved.